nifi-nar-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/ParseNetflowv5.java

```diff
@@ -0,0 +1,258 @@
+
+    public static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(FIELDS_DESTINATION));
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_FAILURE, REL_ORIGINAL, REL_SUCCESS)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        destination = context.getProperty(FIELDS_DESTINATION).getValue();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final OptionalInt portNumber = resolvePort(flowFile);
+        final Netflowv5Parser parser = new Netflowv5Parser(portNumber);
+
+        final byte[] buffer = new byte[(int) flowFile.getSize()];
+        session.read(flowFile, new InputStreamCallback() {
+
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.fillBuffer(in, buffer);
+            }
+        });
+
+        final int processedRecord;
+        try {
+            processedRecord = parser.parse(buffer);
+            getLogger().debug("Parsed {} records from the packet", new Object[] { processedRecord });
+        } catch (Throwable e) {
+            getLogger().error("Parser returned unexpected Exception {} while processing {}; routing to failure", new Object[] { e, flowFile });
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        try {
+            final List<FlowFile> multipleRecords = new ArrayList<>();
+            switch (destination) {
+            case DESTINATION_ATTRIBUTES:
+                final Map<String, String> attributes = new HashMap<>();
+                generateKV(multipleRecords, session, flowFile, attributes, parser, processedRecord);
+                break;
+            case DESTINATION_CONTENT:
+                generateJSON(multipleRecords, session, flowFile, parser, processedRecord, buffer);
+                break;
+            }
+            // Create a provenance event recording the routing to success
+            multipleRecords.forEach(recordFlowFile -> session.getProvenanceReporter().route(recordFlowFile, REL_SUCCESS));
+            session.getProvenanceReporter().route(flowFile, REL_ORIGINAL);
+            // Ready to transfer and commit
+            session.transfer(flowFile, REL_ORIGINAL);
+            session.transfer(multipleRecords, REL_SUCCESS);
+            session.adjustCounter("Records Processed", processedRecord, false);
+            session.commit();
```

Comment:
`AbstractProcessor` will do this for you.

Reply:
Can you please elaborate this comment? What AbstractProcessor will do?

Reply:
Automatically commit the session for you.

Reply:
I tried commenting that line. I am getting error in junit test case..